src/ng/http.js

```diff
@@ -550,6 +550,15 @@ function $HttpProvider() {
      * the transformed value (`function(data, headersGetter, status)`) or an array of such transformation functions,
      * which allows you to `push` or `unshift` a new transformation function into the transformation chain.
      *
+     * <div class="alert alert-warning">
+     * **Note:** Angular does not make a copy of the `data` parameter before it is passed into the `transformRequest` pipeline.
+     * That means changes to the data are not local to the transform function (since Javascript passes objects by reference).
+     * For example, if you are calling `$http.get(url, $scope.myObject)`, and then modify the object's properties in a transformRequest
+     * function, the changes will be reflected on the scope and in any templates where the object is data-bound.
+     * If you are modifying properties, it is therefore recommended to make a copy of the data in the transform function,
+     * or create new object to return.
+     * </div>
+     *
```

Comment:
Is it worth also mentioning that transformRequest and transformResponse functions ought to be "pure", i.e. have no side-effects?